cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java

```diff
@@ -63,4 +71,28 @@ protected void installDependencies() {
     getCommandlet(Java.class).install();
   }
 
+  @Override
+  public void installPlugin(ToolPluginDescriptor plugin, Step step) {
+
+    // In case of plugins with a custom repo url
+    boolean customRepo = plugin.url() != null;
+    List<String> args = new ArrayList<>();
+    args.add("installPlugins");
+    args.add(plugin.id());
+    if (customRepo) {
+      args.add(plugin.url());
+    }
```

Comment:
I have tried to run `... installPlugins <plugin_id> [custom_repo_url]` as described in the docs https://www.jetbrains.com/help/idea/install-plugins-from-the-command-line.html#-ci6zah_17

Unfortunately, I could not make it work. I build my own dummy plugin for testing and hosted a local server where the plugin and the required `updatePlugins.xml` are located, but I get a error message stating that  the plugin could not be found....

Reply:
OK. Got it - so with your new approach you install each plugin by a separate invocation of Intellij.
That might be slower if many plugins need to be installed but should also be KISS and robust.
Also if a plugin installation fails, the error handling should already be in place telling the user via `Step`s which plugin was successfully installed and which not.
We can crate a new story that will even add a progress bar if multiple plugins are to be installed and move it forward by the number of plugins processed.

> maybe the problem is with the local hosting

I assume so. Most probably the URL is just the base-URL of a repository and the full download URL is automatically derived from that and you need to read the documentation from Jetbrains about this.